lib/utils.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+class Utils {
+
+  /**
+   * Used to normalize keys of mixed array and dot-separated string to a single array of undotted strings.
+   *
+   * @param {string|Array} rest (dot-separated) string(s) or array of keys
+   *
+   * @return {Array} The key normalized to an array of simple strings
+   */
+  static normalizeKey(rest) {
+    rest           = Array.isArray(rest) ? rest : Array.prototype.slice.call(arguments);
+    let key        = rest.shift();
+    let normalized = Array.isArray(key) ? Utils.normalizeKey(key) : key.split('.');
+
+    return rest.length === 0 ? normalized : normalized.concat(Utils.normalizeKey(rest));
```

Comment:
Don't really get when `rest.length === 0` here, as we always pass an argument to the function, don't we?

`rest           = Array.isArray(rest) ? rest : Array.prototype.slice.call(arguments);` This will always be greater than `0` in that case. And so we will keep calling `Utils.normalizeKey`.

I guess I am missing something here...